tensorboard/plugins/debugger_v2/debug_data_provider.py

```diff
@@ -283,6 +284,48 @@ def _parse_graph_execution_data_blob_key(blob_key):
     return run, begin, end
 
 
+def graph_op_info_run_tag_filter(run, graph_id, op_name):
+    """Create a RunTagFilter for graph op info.
+
+    Args:
+      run: tfdbg2 run name.
+      graph_id: Debugger-generated ID of the graph. This is assumed to
+        be the ID of the graph that immediately encloses the op in question.
+      op_name: Name of the op in question. (e.g., "Dense_1/MatMul")
+
+    Returns:
+      `RunTagFilter` for the run and range of graph op info.
+    """
+    if not graph_id:
+        raise ValueError("graph_id must not be None or empty.")
+    return provider.RunTagFilter(
+        runs=[run],
+        tags=["%s_%s_%s" % (GRAPH_OP_INFO_BLOB_TAG_PREFIX, graph_id, op_name)],
+    )
+
+
+def _parse_graph_op_info_blob_key(blob_key):
+    """Parse the BLOB key for graph op info.
+
+    Args:
+      blob_key: The BLOB key to parse. By contract, it should have the format:
+       `${GRAPH_OP_INFO_BLOB_TAG_PREFIX}_${graph_id}_${op_name}`
```

Comment:
Apparently this also has `.{$run_name}`?  (Overall I'm not a fan of packing variables into structured strings like this; it seems very brittle.  But I think the approach is water under the bridge-- i.e. I know it was in the design doc a while back etc.-- so I'll try not to complain more about it here :))

Reply:
Right. We are trying to encompass multiple pieces of information in a single blob key. As you mentioned, this is a result of the design that conforms to the DataProvider abstraction, in which there is not a natural fit for efficient look up of 
 individual ops in graphs. So we encode information in blob keys, as an escape hatch. 

This approach is currently okay, because `run_name` is hard-coded to `__default_debugger_run__`. Each logdir currently is mandated to have at most only one run of tfdbg2 data. So there is no ambiguity when run name is included as the tail of the blob key, preceded by a `.`. When we support multiple debugger runs in the same logdir in the future, we can continue enforcing a run name pattern that doesn't contain `.`s.